app/Kconfig

```diff
@@ -194,10 +194,6 @@ config ZMK_BLE_MOUSE_REPORT_QUEUE_SIZE
     int "Max number of mouse HID reports to queue for sending over BLE"
     default 20
 
-config ZMK_BLE_CLEAR_BONDS_ON_START
```

Comment:
Do you think it needs to retain the behavior of only clearing BLE bonds, or could we make it just select the other KConfig option to clear settings on start that you suggested in your other comment?

Reply:
I am a bit on the fence. Other areas we use settings subsystem right now are things like RGB selection, ext power on/off, etc. In the future we also plan to use it for runtime keymap updates.


So... I really think there may be value in both to be supported long term. I get that CircuitPython can muck with the flash region, and having a blunt force "clear all settings" is useful, but I'm not convinced it negates the need for the existing one, as I consider it more.

Thoughts?

Reply:
I was thinking that having a special firmware to clear just bluetooth settings would be unnecessary with something like #2037. The only case that wouldn't cover is fixing pairing of splits, but do we really need to support a special firmware build for that which is separate from the full settings reset?

Reply:
> I was thinking that having a special firmware to clear just bluetooth settings would be unnecessary with something like #2037. The only case that wouldn't cover is fixing pairing of splits, but do we really need to support a special firmware build for that which is separate from the full settings reset?

Repairing splits is *the reason* we originally created the settings reset in the first place. And I do think long term we should have a way to continue to do so that doesn't *also* force a user to lose their runtime keymap they've possible spent a long time updating/creating.

Reply:
I put this back in for now. I'm thinking it should probably still be removed, but only after it's been replaced by a more convenient way to fix pairing for splits--ideally one that doesn't require flashing special firmware.